src/lib/util.c

```diff
@@ -50,6 +50,31 @@ char *string_printf(const char *fmt, ...)
         return ret;
 }
 
+char *string_append(char **original, const char *fmt, ...)
```

Comment:
just reallocate every time you add initrd path. no need to introduce this function.

Reply:
I added it for a future usage avoiding use the same process every time this is required (any way I will modify it)